helix-term/src/commands.rs

```diff
@@ -5203,6 +5143,133 @@ fn select_textobject(cx: &mut Context, objtype: textobject::TextObject) {
+        };
+
+        let selection = doc.selection(view.id).clone().transform(|range| {
+            match ch {
+                'w' => textobject::textobject_word(text, range, objtype, count, false),
+                'W' => textobject::textobject_word(text, range, objtype, count, true),
+                't' => textobject_treesitter("class", range),
+                'f' => textobject_treesitter("function", range),
+                'a' => textobject_treesitter("parameter", range),
+                'c' => textobject_treesitter("comment", range),
+                'T' => textobject_treesitter("test", range),
+                'p' => textobject::textobject_paragraph(text, range, objtype, count),
+                'm' => textobject::textobject_pair_surround_closest(text, range, objtype, count),
+                'g' => textobject_change(range),
+                // TODO: cancel new ranges if inconsistent surround matches across lines
+                ch if !ch.is_ascii_alphanumeric() => {
+                    textobject::textobject_pair_surround(text, range, objtype, ch, count)
+                }
+                _ => range,
+            }
+        });
+        doc.set_selection(view.id, selection);
+    };
+    cx.editor.apply_motion(textobject);
+}
+
+fn select_next_word(cx: &mut Context) {
+    let current_selection = get_current_selection(cx);
+    select_textobject_for_char(cx, 'w', textobject::TextObject::Inside, cx.count());
+    let new_selection = get_current_selection(cx);
+    if selection_eq(current_selection, new_selection) {
+        move_next_word_end(cx);
+        select_textobject_for_char(cx, 'w', textobject::TextObject::Inside, cx.count());
+    }
+}
+
+fn select_next_long_word(cx: &mut Context) {
+    let current_selection = get_current_selection(cx);
+    select_textobject_for_char(cx, 'W', textobject::TextObject::Inside, cx.count());
+    let new_selection = get_current_selection(cx);
+    if selection_eq(current_selection, new_selection) {
+        move_next_long_word_end(cx);
+        select_textobject_for_char(cx, 'W', textobject::TextObject::Inside, cx.count());
+    }
+}
+
+fn select_prev_word(cx: &mut Context) {
+    let current_selection = get_current_selection(cx);
+    select_textobject_for_char(cx, 'w', textobject::TextObject::Inside, cx.count());
+    flip_selections(cx);
+    let new_selection = get_current_selection(cx);
+    if selection_eq(current_selection, new_selection) {
+        move_prev_word_start(cx);
+        select_textobject_for_char(cx, 'w', textobject::TextObject::Inside, cx.count());
+        flip_selections(cx);
+    }
+}
+
+fn select_prev_long_word(cx: &mut Context) {
+    let current_selection = get_current_selection(cx);
+    select_textobject_for_char(cx, 'W', textobject::TextObject::Inside, cx.count());
+    flip_selections(cx);
+    let new_selection = get_current_selection(cx);
+    if selection_eq(current_selection, new_selection) {
```

Comment:
I've implemented these commands using a notion of selection equality, but I'm unsure if the way I've done this is conform best practices

Reply:
Well, at the least the naming should be adjusted according to clippy, but even then I'm unsure.